### PR TITLE
Merak-Compute Database Refactor, Wrappers and Unit Tests

### PR DESCRIPTION
**Pull request by phudtran**
This PR adds code quality improvements for the following

- A wrapper, interface, and unit tests for the go-redis client
- Port over scenario manager db functions to new go-redis client wrapper
- Merak-compute VM and Host entity abstractions and unit tests
- nodeSelector serviceHost=merak for the compute.dev.yaml file

### CODE REVIEW

File: services/datastore/datastore.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+// This is the datastore interface used by merak-compute
+package datastore
+
+import (
+	"context"
+)
+
+type Store interface {
+	Close() error
```
Comment (cj-chung) [inline services/datastore/datastore.go:23]: @phudtran can we move the datastore and it's api to another folder, like under /api?

File: services/datastore/go-redis.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+func getKeys(ctx context.Context, prefix string, store *DB) ([]string, error) {
+	var allkeys []string
+
+	iter := store.Client.Scan(ctx, 0, prefix, 0).Iterator()
+	for iter.Next(ctx) {
+		allkeys = append(allkeys, iter.Val())
+	}
+
```
Comment (cj-chung) [inline services/datastore/go-redis.go:222]: @phudtran how to make sure these functions work?

Comment (phudtran) [reply services/datastore/go-redis.go:222]: I ported these over from the scenario managers database exactly as they were.

File: services/datastore/datastore.go (hunk 1 above)
Comment (phudtran) [reply services/datastore/datastore.go:23]: How about under the common folder?

Comment (cj-chung) [reply services/datastore/datastore.go:23]: @phudtran common folder is ok too.

Comment (phudtran) [reply services/datastore/datastore.go]: updated

**Comment by cj-chung** [comment]
@phudtran Can you write an example or extract the code from your component and let other members know how to use this wrapper?

**Review by cj-chung — APPROVED** [review]
LGTM

**Pull request merged**